int32 node_id
---
int32[] neigh_table
